BiometricLockerTests/BiometricLockerTests.swift

```diff
@@ -0,0 +1,116 @@
+
+        // Since lockers are not namespaced or anything, if you have multiple lockers, they'll all use the same UserDefaults date value.
+        // To ensure we're always starting from scratch, create one and unlock to clear the UserDefaults value. Will improve on this.
+        BiometricLocker(localizedReason: "").unlock()
+    }
+
+    func testLockUnlock() {
+        let expectation = self.expectation(description: "Locking")
+
+        let locker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        XCTAssertFalse(locker.isLocked)
+
+        locker.lock()
+        XCTAssertFalse(locker.isLocked)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4) {
+            XCTAssertTrue(locker.isLocked)
+            expectation.fulfill()
+        }
+
+        self.wait(for: [expectation], timeout: 5)
+    }
+
+    func testLockUnlockImmediatelly() {
+        let locker = BiometricLocker(localizedReason: "")
+        XCTAssertFalse(locker.isLocked)
+
+        locker.lock(.now)
+        XCTAssertTrue(locker.isLocked)
+
+        locker.unlock()
+        XCTAssertFalse(locker.isLocked)
+    }
+
+    func testLockingAfterTimeAllowance() {
+        let expectation = self.expectation(description: "Allowance time lock")
+
+        let locker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        locker.lock(.afterTimeAllowance)
+        XCTAssertFalse(locker.isLocked)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4) {
+            XCTAssertTrue(locker.isLocked)
+            expectation.fulfill()
+        }
+
+        self.wait(for: [expectation], timeout: 5)
+    }
+
+    func testLockingAfterCustomIntervalShorterThanTimeAllowance() {
+        let locker = BiometricLocker(localizedReason: "", withUnlockedTimeAllowance: 3)
+        XCTAssertFalse(locker.isLocked)
+        let expectation = self.expectation(description: "Custom time lock shorter than time allowance")
+
+        locker.lock(.afterTimeInterval(5))
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4) {
+            // despite our time allowance being only 3, it should still be unlocked,
+            // as we've defined a custom time interval when locking.
+            XCTAssertFalse(locker.isLocked)
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 2, execute: {
+                // After a total of 6s, the locker should be locked, according to our custom time interval.
+                XCTAssertTrue(locker.isLocked)
+
+                expectation.fulfill()
+            })
+        }
+
+        self.wait(for: [expectation], timeout: 7)
+    }
+
+    func testLockingAfterCustomInterval() {
+        let locker = BiometricLocker(localizedReason: "")
+        XCTAssertFalse(locker.isLocked)
+        let expectation = self.expectation(description: "Custom time lock")
+
+        locker.lock(.now)
+        XCTAssertTrue(locker.isLocked)
+
+        locker.unlock()
+        XCTAssertFalse(locker.isLocked)
```

Comment:
Why does this need to have a separate locker that locks and unlocks before the main body of the test?

Reply:
Go-go-gone!

Reply:
[addressed]